Add notice about the OTeL collector (#891)
signalfx/integrations: gateway/README.md

**Note:** The relevant code for the project can be found in the <a target="_blank" href="https://github.com/signalfx/gateway">signalfx
/gateway repository</a>.

> **Note:** The SignalFx Gateway is deprecated. You should now deploy the OpenTelemetry Collector and send data from the SignalFx Smart Agent to an OpenTelemetry Collector. See the [OpenTelemetry Collector Transition Guide](https://docs.signalfx.com/en/latest/otel/gateway-otel-migration.html) for more information.

Reviewer: ```suggestion
**Note:** The SignalFx Gateway is deprecated. Deploy the OpenTelemetry Collector and send data from the SignalFx Smart Agent to an OpenTelemetry Collector. See the [OpenTelemetry Collector Transition Guide](https://docs.signalfx.com/en/latest/otel/gateway-otel-migration.html) for more information.
```

Reviewer: we should also link to the new deploy guide that you're working on, traci.

> **Note:** The SignalFx Gateway is being deprecated. You can now deploy the OpenTelemetry Collector and send data from the SignalFx Smart Agent to an OpenTelemetry Collector. See the [OpenTelemetry Collector Transition Guide](https://docs.signalfx.com/en/latest/otel/gateway-otel-migration.html) for more information.

Reviewer 2: ```suggestion
**Note:** The SignalFx Gateway is deprecated. You should now deploy the OpenTelemetry Collector and send data from the SignalFx Smart Agent to an OpenTelemetry Collector. See the [OpenTelemetry Collector Transition Guide](https://docs.signalfx.com/en/latest/otel/gateway-otel-migration.html) for more information.
```
